scripts/hfButler_cpp_clone.py

```diff
@@ -0,0 +1,378 @@
+    hw = uhal.ConnectionManager("file://"+connection).getDevice(str(device))
+    hw.setTimeoutPeriod(10000)
+    obj.mHW = hw
+    obj.podNode = hw.getNode()
+    # Extract info from InfoNode
+    infoNode = obj.podNode.get_info_node()
+    config_info = infoNode.get_firmware_config_info()
+    id_info = infoNode.get_firmware_id_info()
+
+    printDictTable(config_info)
+    printRegTable(id_info)
+    
+    obj.mIdInfo = id_info
+    obj.mConfigInfo = config_info
+
+# -----------------------------------------------------------------------------
+@cli.command('init')
+@click.pass_obj
+def init(obj):
+    obj.podNode.reset()
+    print("\n RESET DONE!!")
+    print("\n Initilisation complete.")
+
+# Command group link------------------------------------------------------------
+@cli.group('link', invoke_without_command=True)
+@click.option('-i', '--ids', default='all', callback=optionValidators.validate_link_ids)
+@click.pass_obj
+def link(obj, ids):
+    """
+    Partition specific commands
+
+    ID: Id of the selected partition
+    """
+    obj.mLinkIds = ids
+    try:
+        obj.mLinkNodes = [obj.podNode.get_link_processor_node(i) for i in ids]
+    except Exception as lExc:
+        click.Abort('Wibulator {} not found in address table'.format(id))
+dpr_mux_choices = {
+    'reset': (0, 0),
+    'playback': (0, 1),
+    'sink': (1, 0),
+    'passthrough': (1, 1)
+}
+# ------------------------------------------------------------------------------
+@link.command('config')
+@click.option('--dr-on/--dr-off', 'dr_on', help='Enable data-reception block', default=None)
+@click.option('--dpr-mux', 'dpr_mux', type=click.Choice(list(dpr_mux_choices.keys()),), help='Configure DPR mux', default=None)
+@click.option('--drop-empty/--keep-empty', 'drop_empty', help='Drop empty hit packets', default=None)
+@click.pass_obj
+def link_config(obj, dr_on, dpr_mux, drop_empty):
+
+    for ln in obj.mLinkNodes:
+        print('>> Link Processor', ln.getId())
+
+        if dpr_mux:
+            dprNode = ln.get_data_router_node().get_dpr_node()
+            print("Configuring DPR mux : ", dpr_mux)
+            dprNode.set_mux_in(dpr_mux_choices[dpr_mux][0], True)
+            dprNode.set_mux_out(dpr_mux_choices[dpr_mux][1], True)
+        
+        if dr_on is not None:
+            print("Configuring data-reception : ", str(dr_on))
+            drNode = ln.get_data_router_node().get_data_reception_node()
+            drNode.getNode("csr.ctrl.en").write(dr_on)
+            drNode.getClient().dispatch()
+
+        if drop_empty is not None:
+            strmArrayNode = ln.get_stream_proc_array_node()
+            strmNode = strmArrayNode.get_stream_proc_node()
+
+            for i in range(obj.mConfigInfo['n_port']):
+                print("Setting drop empty to {} on stream {} ".format(drop_empty, i))
+                strmArrayNode.stream_select(i)
+                strmNode.getNode('csr.ctrl.drop_empty').write(drop_empty)
```

Comment:
StreamProcessorNode::drop_empty() would be better

Reply:
Change made